priv/http_api/backend_swagger.yaml

```diff
@@ -0,0 +1,430 @@
+        200:
+          description: The item was successfully posted for the give user
+        404:
+          description: Failure response because the given user does not exist under this XMPP host or the XMPP host itself does not exist
+    delete:
+      description: Remove an item from a user's roster.
+      parameters:
+        -
+          name: ItemID
+          in: body
+          description: The item to be placed into the user's roster (XMPP hostname passed through the path)
+          required: true
+          schema:
+            title: Item
+            type: object
+            properties:
+              username:
+                type: string
+              XMPPHostname:
+                type: string
+      responses:
+        200:
+          description: The item was successfully removed 
+        404:
+          description: Failure response because the given user does not exist under this XMPP host or the XMPP host itself does not exist
+  /{XMPPHost}/register:
+    parameters:
+        -
+          name: XMPPHost
+          in: path
+          description: The XMPP hostname
+          required: true
+          type: string
+    get:
+      description: Get information about all registered users. This response is **paged** because there may be many users.
+      responses:
+        200:
+          description: Success response listing all registered users
+          schema:
+            title: Resgister
+            type: array
+            items:
+              title: Username
+              type: string
+        404:
+          description: Failure response because the given XMPP host does not exist
+    post:
+      description: Register a user.
+      consumes:
+        - application/json
+      parameters:
+        -
+          name: Credentials
+          in: body
+          description: Credentials for the user to be registered (XMPP hostname passed through the path)
+          required: true
+          schema:
+            title: Credentials
+            type: object
+            properties:
+              username:
+                type: string
+              password:
+                type: string
+      responses:
+        200:
+          description: Register the given entitiy
+        404:
+          description: Failure response because the given XMPP host does not exist
+        409:
+          description: This username is already registered with another entity under this XMPP host
+    delete:
+      description: Cancel a user's registration
+      parameters:
+        -
+          name: Username
+          in: body
+          description: The username part of a JID (XMPP hostname passed through the path)
+          required: true
+          schema:
+            title: username
+            type: string
+      responses:
+        200:
+          description: Cancelation successful
+        404:
+          description: Could not find the given user under the XMPP host
+  /{XMPPHost}/session:
```

Comment:
Probably plural would be better here - `sessions`. Basically it looks to me that all resources should be in plural form.